internal/api/hcpopenshiftcluster.go

```diff
@@ -124,5 +124,8 @@ func NewDefaultHCPOpenShiftCluster() *HCPOpenShiftCluster {
 				},
 			},
 		},
+		Identity: arm.Identity{
+			UserAssignedIdentities: make(map[string]*arm.UserAssignedIdentity),
+		},
```

Comment:
I think a better place to create the map might be in [normalizeIdentityUserAssignedIdentities](https://github.com/Azure/ARO-HCP/blob/026ce1cfdb9eb0b437f75b149b3a9e69f117e988/internal/api/v20240610preview/hcpopenshiftclusters_methods.go#L547-L556).

Reply:
I'm fine with that